watertap/tests/test_pure_water_pH.py

```diff
@@ -569,15 +540,21 @@ def test_scaling_equilibrium(self, equilibrium_reactions_config):
             model.fs.unit.control_volume.properties_in[0.0].scaling_factor, Suffix
         )
 
-        # When using equilibrium reactions, there are another set of scaling factors calculated
+    def test_equilibrium_extra_scale_factors(self, model_scaling, variant: Variant):
+        """
+        When using equilibrium reactions, there are another set of scaling factors calculated
+        """
+        if not variant.is_equilibrium:
+            pytest.skip("Only applicable for equilibrium reaction")
+
```

Comment:
Are we ok with doing this in this way? This will show up in the pytest run as skipped tests (and will always show up as skipped tests).

Reply:
For me the semantics of skipping tests should be that some external condition is not present, but if the conditions are all controlled by the test code itself a simple if/then block would suffice.

Reply:
I guess I have a more "liberal" view of `pytest.skip()`, as in it can be used to express the concept of "this test function does not apply to this context", where the context includes both external conditions (`if sys.platform == "win32": pytest.skip("This test cannot run on Windows")`) and internal data, i.e. fixture values (such as in this case). 

Consequently, in this view, having skipped tests in the output is normal: in some cases, factoring tests for ease of parametrization might result in test-parameter value combinations that don't make sense domain-wise (such as here), and using `pytest.skip()` instead of a "silent" `if` can be a way of making this more explicit in the output.

However, of courses, there are many cases where the line is blurred and both option make sense implementation-wise, in which case I think we should go with the one that makes the most sense domain-wise. In this case, if the assertion(s) to perform for inherent reactions is more of an additional check than a separate test altogether, it makes sense to have a single test function and use a simple `if` statement to bypass the inherent reaction `assert`s for the equilibrium case.